pattern/Conjunction.java

```diff
@@ -114,8 +118,24 @@ public void bound(Map<Reference.Name, Either<Label, byte[]>> bounds) {
             if (var.id().isName() && bounds.containsKey(var.id().reference().asName())) {
                 Either<Label, byte[]> boundVar = bounds.get(var.id().reference().asName());
                 if (var.isType() != boundVar.isFirst()) throw GraknException.of(CONTRADICTORY_BOUND_VARIABLE, var);
-                else if (var.isType()) var.asType().label(boundVar.first());
-                else if (var.isThing()) var.asThing().iid(boundVar.second());
+                else if (var.isType()) {
+                    Optional<LabelConstraint> existingLabel = var.asType().label();
+                    if (existingLabel.isPresent() && !existingLabel.get().properLabel().equals(boundVar.first())) {
+                        var.setSatisfiable(false);
+                        this.setSatisfiable(false);
+                    } else if (!existingLabel.isPresent()) {
+                        var.asType().label(boundVar.first());
+                        var.asType().setResolvedTypes(set(boundVar.first()));
+                    }
+                } else if (var.isThing()) {
+                    Optional<IIDConstraint> existingIID = var.asThing().iid();
+                    if (existingIID.isPresent() && !Arrays.equals(existingIID.get().iid(), (boundVar.second()))) {
+                        var.setSatisfiable(false);
+                        this.setSatisfiable(false);
+                    } else {
+                        var.asThing().iid(boundVar.second());
+                    }
+                }
```

Comment:
when binding an answer to a cloned conjunction, it may be that there's a Pre-existing Label or IID. Here we detect contradictions and set the `isSatisfiable` flag if needed. We also override the resolved types when binding a label